configure.sh

```diff
@@ -17,57 +17,101 @@ PRIVATE_PUSHWOOSH_PROPERTIES="$BASE_PATH/android/pushwoosh.properties"
 PRIVATE_LIBNOTIFY_PROPERTIES="$BASE_PATH/android/libnotify.properties"
 PRIVATE_NETWORK_CONFIG="$BASE_PATH/android/res/xml/network_security_config.xml"
 SAVED_PRIVATE_REPO_FILE="$BASE_PATH/.private_repository_url"
+SAVED_PRIVATE_BRANCH_FILE="$BASE_PATH/.private_repository_branch"
 TMP_REPO_DIR="$BASE_PATH/.tmp.private.repo"
 
-# TODO: Remove these lines when XCode project is finally generated by CMake. 
-cd $BASE_PATH/3party/boost/
-./bootstrap.sh
-./b2 headers
-cd $BASE_PATH
-
-if [ -f "$SAVED_PRIVATE_REPO_FILE" ]; then
-  PRIVATE_REPO=`cat "$SAVED_PRIVATE_REPO_FILE"`
-  echo "Using stored private repository URL: $PRIVATE_REPO"
-else
-  echo "If you are developer from MAPS.ME team, please specify a private repository url here."
-  echo "If not [yet :)], then just press Enter."
-  echo -n "> "
-  read PRIVATE_REPO
-  if [ -z "$PRIVATE_REPO" ]; then
-    echo "Initializing repository with default values in Open-Source mode."
-    cat "$DEFAULT_PRIVATE_HEADER" > "$PRIVATE_HEADER"
-    cat "$DEFAULT_PRIVATE_CAR_MODEL_COEFS" > "$PRIVATE_CAR_MODEL_COEFS"
-    echo 'ext {
+setup_opensource() {
+  echo "Initializing repository with default values in Open-Source mode."
+  cat "$DEFAULT_PRIVATE_HEADER" > "$PRIVATE_HEADER"
+  cat "$DEFAULT_PRIVATE_CAR_MODEL_COEFS" > "$PRIVATE_CAR_MODEL_COEFS"
+  echo 'ext {
   spropStoreFile = "../tools/android/debug.keystore"
   spropStorePassword = "12345678"
   spropKeyAlias = "debug"
   spropKeyPassword = "12345678"
 }
 ' > "$PRIVATE_PROPERTIES"
 
-    echo 'appId=XXXXX
+  echo 'appId=XXXXX
 projectId=00000000
 ' > "$PRIVATE_LIBNOTIFY_PROPERTIES"
 
-    echo 'apiSecret=0000000000000000000000000000000000000000000000000000000000000000
+  echo 'apiSecret=0000000000000000000000000000000000000000000000000000000000000000
 apiKey=0000000000000000000000000000000000000000
 ' > "$PRIVATE_FABRIC_PROPERTIES"
-    echo 'pwAppId=XXXXX
+  echo 'pwAppId=XXXXX
 pwProjectId=A123456789012
 ' > "$PRIVATE_PUSHWOOSH_PROPERTIES"
-    echo '<?xml version="1.0" encoding="utf-8"?>
+  echo '<?xml version="1.0" encoding="utf-8"?>
 <network-security-config/>
 ' > "$PRIVATE_NETWORK_CONFIG"
-    exit
+}
+
+setup_private() {
+  if git clone --depth 1 --no-single-branch "$PRIVATE_REPO" "$TMP_REPO_DIR"; then
+    echo "$PRIVATE_REPO" > "$SAVED_PRIVATE_REPO_FILE"
+    echo "$PRIVATE_BRANCH" > "$SAVED_PRIVATE_BRANCH_FILE"
+    echo "Saved private repository url '$PRIVATE_REPO' to '$SAVED_PRIVATE_REPO_FILE'"
+    echo "Saved private branch '$PRIVATE_BRANCH' to '$SAVED_PRIVATE_BRANCH_FILE'"
+    (cd $TMP_REPO_DIR && git checkout $PRIVATE_BRANCH)
+    rm -rf "$TMP_REPO_DIR/.git" "$TMP_REPO_DIR/README.md"
+    cp -Rv "$TMP_REPO_DIR"/* "$BASE_PATH"
+    rm -rf "$TMP_REPO_DIR"
+    echo "Private files have been updated."
+  fi
+}
+
+ARGS_PRIVATE_REPO=${1-}
```

Comment:
maybe add usage() or at least add the override hierarchy comment to this file?

Reply:
usage() is not really an option, since even without parameters it should work in opensource mode.
Added comment block.